protocol-units/da/movement/protocol/client/src/http1.rs

```diff
@@ -0,0 +1,65 @@
+use hyper_util::client::legacy::connect::HttpConnector;
+use hyper_util::client::legacy::Client;
+use hyper_util::rt::TokioExecutor;
+use movement_da_light_node_proto::light_node_service_client::LightNodeServiceClient;
+use tonic_web::GrpcWebClientLayer;
+use tonic_web::{GrpcWebCall, GrpcWebClientService};
+
+#[derive(Debug, Clone)]
+pub struct Http1 {
+	client: LightNodeServiceClient<
+		GrpcWebClientService<
+			Client<
+				HttpConnector,
+				GrpcWebCall<
+					http_body_util::combinators::UnsyncBoxBody<bytes::Bytes, tonic::Status>,
+				>,
+			>,
+		>,
+	>,
+}
+
+impl Http1 {
+	/// Tries to connect to a light node service using the given connection string.
+	pub fn try_new(connection_string: &str) -> Result<Self, anyhow::Error> {
+		let client = hyper_util::client::legacy::Client::builder(TokioExecutor::new()).build_http();
+
+		let svc = tower::ServiceBuilder::new().layer(GrpcWebClientLayer::new()).service(client);
+
+		let client = LightNodeServiceClient::with_origin(svc, connection_string.try_into()?);
+
+		Ok(Http1 { client })
+	}
+
+	/// Returns a reference to the client.
+	pub fn client(
+		&self,
+	) -> &LightNodeServiceClient<
+		GrpcWebClientService<
+			Client<
+				HttpConnector,
+				GrpcWebCall<
+					http_body_util::combinators::UnsyncBoxBody<bytes::Bytes, tonic::Status>,
+				>,
+			>,
+		>,
+	> {
```

Comment:
`LightNodeServiceClient` does not ship with a companion trait. `tonic` directly templates all of the methods and does not actually `impl LightNodeService` on the client.

Reply:
Cribbing from `tonic::transport::Channel`, this could be:

```rust
pub type Body = UnsyncBoxBody<Bytes, tonic::Status>;

impl Http1 {
	pub fn client(
		&self,
	) -> &LightNodeServiceClient<
		impl Service<Request<Body>, Response = Response<Body>>,
	> {
            // ...
        }
```

Reply:
OK, it's not as easy as it sounds, and it might have to do with use of the legacy `Client` from hyper-util.